src/Components/Menu/Menu.js

```diff
@@ -1,31 +1,89 @@
-import React, { useEffect, useState} from 'react';
+import React, { useEffect, useState } from 'react';
 import { useStore } from '../App/App.js';
-import { getFolders, postFolder } from '../../apiCalls/apiCalls'
+import { getFolders, postFolder, postPalette } from '../../apiCalls/apiCalls'
 import './Menu.scss';
 import { Link } from 'react-router-dom';
 import shuffle from '../../Images/001-random.svg';
 import heart from '../../Images/002-heart.svg'
 import folderIcon from '../../Images/003-folder.svg'
 
 export const Menu = () => {
+  const [ newFolder, setNewFolder ] = useState({})
   const [ folder, setFolder ] = useState({});
   const [ folders, setFolders ] = useState([]);
+  const [ paletteName, setPaletteName ] = useState('');
+  const [ error, setError ] = useState('')
+  const [ paletteSuccess, setPaletteSuccess ] = useState('')
+  const [ folderSuccess, setFolderSuccess ] = useState('')
   const { state, dispatch } = useStore();
 
-  let handleClick = () => {
-    postFolder(folder)
+  let evaluateInput = (e) => {
+    if (e.target.name === 'choose_folder' && e.target.value === 'choose existing folder') {
+      setError('select a folder')
+    }
+    if (e.target.name === 'heart' && !paletteName) {
+      setError('enter a palette name')
+    }
+    if (e.target.name === 'create_folder' && !newFolder.folder_name) {
+      setError('enter a folder name')
+    }
+    if (e.target.name === 'create_folder' && newFolder.folder_name) {
+      handleFolderPost()
+    }
+    if (e.target.name === 'heart' && !folder.folder_name) {
+      setError('select a folder')
+    }
+    if (e.target.name === 'heart' && paletteName && folder.folder_name) {
+      handlePalettePost()
+    }
+  }
+
+  let handlePalettePost = () => {
+    let folderName = folder.folder_name
+    let matchingFolder = folders.find(folder => folder.folder_name === folderName)
+  
+    let palette = {
+      paletteName: paletteName,
+      colors: [
+        state[0].color,
+        state[1].color,
+        state[2].color,
+        state[3].color,
+        state[4].color
+      ]
+    }
+
```

Comment:
I have been noticing on my end that the newly posted palettes are coming through blank, and pretty sure it can be solved by adding a # before each of these colors.